File: palindromic-substrings/palindromic-substrings.cpp
class Solution {
public:

    int fun(string s,int l,int r) {
        int count=0;
        while(l>=0 && r<s.length() && s[l]==s[r])
        {
            count++;
            l--;
            r++;
        }
        return count;
    }

    int countSubstrings(string s) {
        int n = s.length();
        int ans=0;
        for(int i=0;i<n;i++)
        {
            ans+=fun(s,i,i);
            ans+=fun(s,i,i+1);
        }
        return ans;
    }
};

File: 647-palindromic-substrings/palindromic-substrings.java
class Solution {
    public int countSubstrings(String s) {
        char[] ch = s.toCharArray();
        int n = ch.length;

        int ans = 0;

        boolean[][] dp = new boolean[n][n];

        for(int i=0;i<n;i++) dp[i][i]=true;
        ans+=n;

        for(int size=2;size<=n;size++)
        {
            for(int l=0;l<=n-size;l++)
            {
                int r = l+size-1;
                if(size==2)
                {
                    if(ch[l]==ch[r]) dp[l][r]=true;
                }
                else
                {
                    if(dp[l+1][r-1]==true && ch[l]==ch[r]) dp[l][r]=true;
                }
                if(dp[l][r]==true) ans++;
            }
        }
        return ans;
    }
}